test/rpc-integration/resources/simplify.k

```diff
@@ -1,8 +1,28 @@
 module SIMPLIFY
   imports INT
+  imports BOOL
   imports ID-SYNTAX
 
   syntax Int ::= f ( Int ) [function, no-evaluators] // a non-concrete value
+               | g ( Int ) [function, klabel("g"), symbol] // for simplification purposes
+
+  syntax Bool ::= p1 ( Int ) [function, total, no-evaluators, klabel("p1"), symbol]
+                | p2 ( Int ) [function, total, no-evaluators, klabel("p2"), symbol]
+                | p3 ( Int ) [function, total, klabel("p3"), symbol]
+
+  // an indirect simplifiation loop
+  rule [p1-if-p2]: p1(X) => true requires p2(X) [simplification]
+  rule [p2-if-p1]: p2(X) => true requires p1(X) [simplification]
+  // simplification of terms involving p1(_someInt) or p2(_someInt) will
+  // enter a loop of p1, p2, p1, p2... until depth 100 is reached, and then
+  // bail out in booster. Simplification in kore-rpc _loops to infinity_.
```

Comment:
This can be tested with `booster-dev` - it recurses 100 times and then terminates with `TooManyIterations`. There is no loop being detected because it does not actually ever change the term. To see the loop detection kick in, we could create a test that uses `Int` arithmetic (and an llvm dy.lib) or some other simplifications  that happen before we enter the loop.